src/hotspot/share/interpreter/linkResolver.cpp

```diff
@@ -972,15 +972,22 @@ void LinkResolver::resolve_field(fieldDescriptor& fd,
 
   if (resolved_klass == NULL) {
     ResourceMark rm(THREAD);
-    THROW_MSG(vmSymbols::java_lang_NoSuchFieldError(), field->as_C_string());
+    THROW_MSG(vmSymbols::java_lang_NoSuchFieldError(),
+        err_msg("No resolved class, field '%s %s'",
+                type2name(Signature::basic_type(sig->char_at(0))),
```

Comment:
Isn't this going to report "object" for all non-primitive types?

Reply:
You're absolutely right, this would output "object" for all references and "array" for all array types. Ideally this should be more detailed about these types.

Reply:
There's no test case for this condition.